File: Programing_Basics_with_JS/conditionalStatementsMoreExercises/01.PipesInPool.js
function pipesInPool(input) {
    let poolVolume = Number(input[0]);
    let firstPipeFlowRate = Number(input[1]);
    let secondPipeFlowRate = Number(input[2]);
    let hoursWithoutWorker = Number(input[3]);

    let firstPipeLiters = firstPipeFlowRate * hoursWithoutWorker;
    let secondPipeLiters = secondPipeFlowRate * hoursWithoutWorker;
    let poolWater = firstPipeLiters + secondPipeLiters;
    let poolWaterPercent = poolWater / poolVolume * 100;
    let firstPipeLitersPercent = firstPipeLiters / poolWater * 100;
    let secondPipeLitersPercent = secondPipeLiters / poolWater * 100;
    let overFlowsLiters = poolWater - poolVolume;

    if (poolWater <= poolVolume) {
        console.log(`The pool is ${poolWaterPercent.toFixed(2)}% full. Pipe 1: ${firstPipeLitersPercent.toFixed(2)}%. Pipe 2: ${secondPipeLitersPercent.toFixed(2)}%.`);
    } else {
        console.log(`For ${hoursWithoutWorker.toFixed(2)} hours the pool overflows with ${overFlowsLiters.toFixed(2)} liters.`);
    }
}

pipesInPool(['1000', '100', '120', '3'])
pipesInPool(['100', '100', '100', '2.5'])

File: Programming_Basics_with_Python/conditional_statements_more_exercise/01._pipes_in_pool.py
def pool_condition(pool_capacity, first_pipe, second_pipe, missing_hours):
    first_pipe_water = first_pipe * missing_hours
    second_pipe_water = second_pipe * missing_hours
    total_water_from_two_pipes = first_pipe_water + second_pipe_water
    pool_water_percent = (total_water_from_two_pipes / pool_capacity) * 100
    first_pipe_water_percent = (first_pipe_water / total_water_from_two_pipes) * 100
    second_pipe_water_percent = (second_pipe_water / total_water_from_two_pipes) * 100
    overflows_water = abs(pool_capacity - total_water_from_two_pipes)

    if pool_capacity >= total_water_from_two_pipes:
        return f'The pool is {pool_water_percent:.2f}% full. Pipe 1: {first_pipe_water_percent:.2f}%. Pipe 2: {second_pipe_water_percent:.2f}%."'
    else:
        return f'For {missing_hours:.2f} hours the pool overflows with {overflows_water:.2f} liters.'


pool_capacity = int(input())
first_pipe = int(input())
second_pipe = int(input())
missing_hours = float(input())

print(pool_condition(pool_capacity, first_pipe, second_pipe, missing_hours))